modules/programs/hexchat.nix

```diff
@@ -0,0 +1,324 @@
+              useGlobalUserInformation = false;
+              forceSSL = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+
+    settings = mkOption {
+      default = null;
+      description = ''
+        Configuration for <filename>~/.config/hexchat/hexchat.conf</filename>, see
+        <link xlink:href="https://hexchat.readthedocs.io/en/latest/settings.html#list-of-settings"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };
+      '';
+      type = nullOr (attrsOf str);
+    };
+
+    overwriteConfigFiles = mkOption {
+      type = nullOr bool;
+      description = ''
+        Enables overwritting HexChat configuration files (hexchat.conf, servlist.conf).
+        Any existing HexChat configuration will be lost.
+        Certify to back-up any previous configuration before enabling this.
+
+        Enabling this setting is recommended, because everytime HexChat application is
+        closed it overwrites Nix/Home Manager provided configuration files, causing:
+        1. Nix/Home Manager provided configuration to be out of sync with actual active HexChat configuration.
+        2. Blocking Nix/Home  Manager updates until configuration files are manually removed.
+      '';
+      default = false;
+    };
+
+    theme = mkOption {
+      default = null;
+      description = ''
+        Theme package for HexChat.
+        Expects a derivation containing decompressed theme files.
+        '.hct' file format requires unzip decompression, as seen in example.
+      '';
+      example = literalExample ''
+        stdenv.mkDerivation rec {
+          name = "hexchat-theme-MatriY";
+          buildInputs = [ pkgs.unzip ];
+          src = fetchurl {
+              url = "https://dl.hexchat.net/themes/MatriY.hct";
+              sha256 = "sha256-ffkFJvySfl0Hwja3y7XCiNJceUrGvlEoEm97eYNMTZc=";
+          };
+          unpackPhase = "unzip ''${src}";
+          installPhase = "cp -r . $out";
+        };
+      '';
+      type = nullOr package;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.hexchat ];
+    xdg.configFile."hexchat" = mkIf (cfg.theme != null) {
+      source = cfg.theme;
+      recursive = true;
+    };
+
+    xdg.configFile."hexchat/hexchat.conf" = mkIf (cfg.settings != null) {
+      force = cfg.overwriteConfigFiles;
+      text = concatMapStringsSep "\n" (x: x + " = " + cfg.settings.${x})
+        (attrNames cfg.settings);
+    };
+
+    xdg.configFile."hexchat/servlist.conf" = mkIf (cfg.channels != null)
+      (if attrNames cfg.channels == [ ] then {
+        text = "";
```

Comment:
Why not change the default value to an empty set if there's a check for an empty set afterwards?